### PR TITLE
feat: Display warning when output of `list` subcommands are truncated in cloud CLI

### PR DESCRIPTION
**Pull request by BraedonLeonard**
closes: https://github.com/meltano/meltano/issues/7588

### CODE REVIEW

**Comment by BraedonLeonard** [comment]
The failing checks are due to 2 bugs:
- https://github.com/meltano/meltano/issues/7812
- https://github.com/meltano/meltano/issues/7813

File: src/cloud-cli/meltano/cloud/cli/base.py [hunk 1]
```diff
@@ -38,6 +46,119 @@ def wrapper(*args, **kwargs):
...
+    """
+    List of items, along with whether or not that list is truncated.
+
+    Used to store lists that are to be printed by `print_formatted_list`.
+    """
+
+    def __init__(self, items: list[T] | None = None, truncated: bool = False):
+        """Initialize items to an empty list and truncated to False."""
+        self.items: list[T] = items if items is not None else []
+        self.truncated: bool = truncated
```
Comment (WillDaSilva) [inline src/cloud-cli/meltano/cloud/cli/base.py:62]:
```suggestion
from dataclasses import dataclass, field

@dataclass
class LimitedResult(t.Generic[T]):
    """
    List of items, along with whether or not that list is truncated.

    Used to store lists that are to be printed by `print_formatted_list`.
    """
    items: list[T] = field(default_factory=list)
    truncated: bool = False
```

File: src/cloud-cli/tests/cli/test_base.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    test_case(max_items=5, limit=6, max_page_size=10)
+
+    # Case: Limit less than number of items
+    test_case(max_items=5, limit=4, max_page_size=10)
+
+    # Case: Limit greater than number of items, multi-page
+    test_case(max_items=15, limit=16, max_page_size=10)
+
+    # Case: Limit less than number of items, multi-page
+    test_case(max_items=15, limit=14, max_page_size=10)
```
Comment (WillDaSilva) [inline src/cloud-cli/tests/cli/test_base.py:56]: This seems like a good place to use [`pytest.mark.parameterize`](https://docs.pytest.org/en/7.2.x/how-to/parametrize.html).

The suggestion below is about what that would look like. I haven't tested the suggestion.

```suggestion
from __future__ import annotations

import pytest

from meltano.cloud.cli.base import get_paginated, run_async

@pytest.mark.parametrize(('max_items', 'limit', 'max_page_size'), (
    # Case: Page size, limit, and number of items all equal
    (10, 10, 10),
    # Case: Limit less than number of items, equal to page size
    (11, 10, 10),
    # Case: Limit greater than number of items
    (5, 6, 10),
    # Case: Limit less than number of items
    (5, 4, 10),
    # Case: Limit greater than number of items, multi-page
    (15, 16, 10),
    # Case: Limit less than number of items, multi-page
    (15, 14, 10),
))
def test_get_paginated(max_items: int, limit: int, max_page_size: int):
    async def paged_func(page_size: int, page_token: str, max_items: int):
        if not page_token:
            return {
                "results": list(range(min(page_size, max_items))),
                "pagination": {"next_page_token": str(page_size)}
                if max_items > page_size
                else None,
            }
        return {
            "results": list(
                range(int(page_token), min(int(page_token) + page_size, max_items)),
            ),
            "pagination": None,
        }

    results = run_async(get_paginated)(
        lambda page_size, page_token: paged_func(
            page_size,
            page_token,
            max_items=max_items,
        ),
        limit=limit,
        max_page_size=max_page_size,
    )

    assert results.items == list(range(min(limit, max_items)))
    assert results.truncated == (limit < max_items)
```

**Review by WillDaSilva — COMMENTED** [review]
Thanks for DRY-ing up this code @BraedonLeonard, and for the clean implementation.

I've manually tested it with several cases, and found no issues. The warning is printed to stderr, and so is easy to separate from the regular output.

File: src/cloud-cli/tests/cli/test_base.py (hunk 2 above)
Comment (BraedonLeonard) [reply src/cloud-cli/tests/cli/test_base.py:56]: Good call 👍

**Review by WillDaSilva — APPROVED** [review]

**Pull request merged**